core/src/main/java/org/apache/struts2/dispatcher/Dispatcher.java

```diff
@@ -304,6 +310,20 @@ public boolean isHandleException() {
         return handleException;
     }
 
+    /**
+     * Modify state of StrutsConstants.STRUTS_PRINTSTACKTRACE_ONEXCEPTION setting.
+     * 
+     * @param printStackTraceOnException 
+     */
+    @Inject(value = StrutsConstants.STRUTS_PRINTSTACKTRACE_ONEXCEPTION, required = false)
```

Comment:
I think @aleksandr-m meant (or at least I would like) to just revert [accidentally added `e.print...`](https://issues.apache.org/jira/browse/WW-5028?focusedCommentId=16808380&page=com.atlassian.jira.plugin.system.issuetabpanels:comment-tabpanel#comment-16808380) line. Or as a last resort we can just replace it with a `LOG.debug`. IMHO more complexity (e.g. enable user to change it via configuration) isn't essential here.

Reply:
Hello @yasserzamani .

Thanks for the feedback.

Having a control flag might not be essential, but having control over Dispatcher producing stacktraces (whether to stdout or a log) is convenient and seems to provide a solution in line with what the JIRA opener was seeking.

I think it's fair to offer users the ability to decide, via a control flag, if they want to see/track these `Dispatcher` errors (sometime Log level-separation isn't flexible enough).

Once the others give their feedback the PR can be updated to go with whatever the consensus is.

Reply:
I try to represent my concern in another way :) I think anyone who agree a control flag should answer then why we don't have such logic for other classes and why only `Dispatcher` is specialized for such user control on exceptions via configuration?

I understand your concern and now I can have a better recommendation below instead of my previous recommendation `LOG.debug`:
```diff
Index: core/src/main/java/org/apache/struts2/dispatcher/DefaultDispatcherErrorHandler.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- core/src/main/java/org/apache/struts2/dispatcher/DefaultDispatcherErrorHandler.java	(revision 68044e741bf2566c96c876265ce99f9a851f96f7)
+++ core/src/main/java/org/apache/struts2/dispatcher/DefaultDispatcherErrorHandler.java	(date 1551013035271)
@@ -101,7 +101,7 @@
     }
 
     protected void handleErrorInDevMode(HttpServletResponse response, int code, Exception e) {
-        LOG.debug("Exception occurred during processing request: {}", e.getMessage(), e);
+        LOG.warn("Exception occurred during processing request: {}", e.getMessage(), e);
         try {
             List<Throwable> chain = new ArrayList<>();
             Throwable cur = e;
```
i.e. warn in `devMode` which I think is enough.